### PR TITLE
VZ-10172: more meaningful compute instance names

### PR DESCRIPTION
**Pull request by marcogelle**
Includes the OCNE cluster "display name" in the names of the compute instances created for the control plane and the node pools.

Example:
The user-specified cluster name is `ui-cluster-dollar-display-name`. The user-specified node pool name is `pool-6f914`. This is how the instance names appear on the Rancher UI.
Screenshot 2023-07-10 at 10 35 47 AM

### CODE REVIEW

File: pkg/templates/ocimachinetemplate.goyaml [hunk 1]
```diff
@@ -9,7 +9,7 @@ items:
   - apiVersion: infrastructure.cluster.x-k8s.io/v1beta2
     kind: OCIMachineTemplate
     metadata:
-      name:  {{.Name}}-{{$.NodePoolHash}}
+      name:  {{$.DisplayName}}-{{.Name}}-{{$.NodePoolHash}}
```
Comment (anders-swanson) [inline pkg/templates/ocimachinetemplate.goyaml:12]: You can use `{{$.DisplayName}}-{{$.NodePoolHash}}`. I don't think {{.Name}} needs to be in there

Comment (marcogelle) [reply pkg/templates/ocimachinetemplate.goyaml:12]: Sure, I can change that

**Comment by marcogelle** [comment]
> What happens during the update scenario? This is going to create new machinepools, right?

I'm unaware how changing the name here would change the update functionality. I can manually test this out.

**Comment by marcogelle** [comment]
I tested the update scenario with these changes locally. I started with a single node pool cluster and performed updates to add and remove a node pool. It seems to work fine.
donut2-pre-update
donut2-update1
donut2-update2

File: pkg/templates/ocimachinetemplate.goyaml (hunk 1 above)
Comment (marcogelle) [reply pkg/templates/ocimachinetemplate.goyaml]: I made this change.

**Review by anders-swanson — APPROVED** [review]

**Review by bobdonat — DISMISSED** [review]

**Comment by marcogelle** [comment]
Other update scenario: start with node pools created from old kontainer-driver, then edit the cluster with this new kontainer driver.
In this example, the update performed is adding one node pool. Screenshots are before and after the update.
Result: The control plane and worker nodes are all updated following the new naming convention.
update-donut-preupdate
update-donut-update1

**Review by anders-swanson — DISMISSED** [review]

**Review by anders-swanson — APPROVED** [review]

**Pull request merged**